packages/engine/src/gltf/GLTFLoaderFunctions.ts

```diff
@@ -1545,7 +1545,21 @@ const loadNode = async (options: GLTFParserOptions, nodeIndex: number) => {
 
   return nodeEntity
 }
+const loadMaterialGLTF = async (options: GLTFParserOptions, sceneIndex: number) => {
+  const json = Array.isArray(options.document) ? options.document[0] : options.document
 
+  // load deltas into state before anything else
+  const deltas = json.extensions?.[SCENE_DELTA_EXTENSION_NAME] as SceneDeltaRegistry | null
+  if (deltas) {
+    const parsedDeltas = parseStorageProviderURLs(deltas)
+    getMutableState(SceneDeltaState).merge(parsedDeltas)
+  }
+  DependencyCache.set(options.url, new Map())
+  const pending = [] as Promise<Entity>[]
+  pending.push(getDependency(options, 'material', 0) as Promise<Entity>)
+  await Promise.all(pending)
+  getComponent(options.entity, GLTFComponent).body = null
```

Comment:
Don't call getDependency. Just loop through the gltf document's "materials" field and call "loadMaterial" for each index